exec/java-exec/src/main/java/org/apache/drill/exec/planner/cost/DrillRelMdSelectivity.java

```diff
@@ -173,6 +177,58 @@ private double getScanSelectivityInternal(DrillTable table, RexNode predicate, L
     return (sel > 1.0) ? 1.0 : sel;
   }
 
+  private double computeEqualsSelectivity(DrillTable table, RexNode orPred, List<String> fieldNames) {
+    String col = getColumn(orPred, fieldNames);
+    if (col != null) {
+      if (table.getStatsTable() != null
+              && table.getStatsTable().getNdv(col) != null) {
+        return 1.00 / table.getStatsTable().getNdv(col);
+      }
+    }
+    return guessSelectivity(orPred);
+  }
+
+  private double computeIsNullSelectivity(DrillTable table, RexNode orPred, List<String> fieldNames) {
+    String col = getColumn(orPred, fieldNames);
+    if (col != null) {
+      if (table.getStatsTable() != null
+          && table.getStatsTable().getNNRowCount(col) != null) {
+        // Cap selectivity between {1/Total_Rows, Calcite Guess}
+        return Math.min(Math.max(1.0/table.getStatsTable().getRowCount(),
```

Comment:
I missed the 1/getRowCount()...why not just min( (1 - NNRowCount/totalRowCount),  guessSelectivity) ?

Reply:
That was just to prevent the selectivity to become 0. I vaguely remember running into planning issues with 0 selectivity - maybe I am being paranoid. I will remove it for now.